addons/pos_sale/models/pos_order.py

```diff
@@ -134,5 +134,7 @@ def _order_line_fields(self, line, session_id=None):
         if vals.get('sale_order_origin_id', False):
             vals['sale_order_origin_id'] = vals['sale_order_origin_id']['id']
         if vals.get('sale_order_line_id', False):
-            vals['sale_order_line_id'] = vals['sale_order_line_id']['id']
+            #We need to make sure the order line has not been deleted while the order was being handled in the PoS
+            order_line = self.env['sale.order.line'].search([('id', '=', vals['sale_order_line_id']['id'])], limit=1)
```

Comment:
Can you try the `exists` method?

Reply:
I don't know if it's really usefull here, what would be the difference ? We still need to do the search with the id of the sale_order_line